test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,263 @@
+		[]string{
+			"--audit-dynamic-configuration",
+			"--feature-gates=DynamicAuditing=true",
+			"--runtime-config=auditregistration.k8s.io/v1alpha1=true",
+		},
+		framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	eventList1 := &atomic.Value{}
+	eventList1.Store(auditinternal.EventList{})
+	eventList2 := &atomic.Value{}
+	eventList2.Store(auditinternal.EventList{})
+
+	// start test servers
+	server1 := httptest.NewServer(buildTestHandler(t, eventList1))
+	defer server1.Close()
+	server2 := httptest.NewServer(buildTestHandler(t, eventList2))
+	defer server2.Close()
+
+	scheme := runtime.NewScheme()
+	auditinstall.Install(scheme)
+	serializer := json.NewSerializer(json.DefaultMetaFactory, scheme, scheme, false)
+	encoder := audit.Codecs.EncoderForVersion(serializer, auditv1.SchemeGroupVersion)
+
+	// check that servers are healthy
+	err := testServerHealthCheck(server1.URL, encoder)
+	require.NoError(t, err)
+	eventList1.Store(auditinternal.EventList{})
+
+	err = testServerHealthCheck(server2.URL, encoder)
+	require.NoError(t, err)
+	eventList2.Store(auditinternal.EventList{})
+
+	kubeclient, err := kubernetes.NewForConfig(result.ClientConfig)
+	require.NoError(t, err)
+
+	sink1 := auditregv1alpha1.AuditSink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test1",
+		},
+		Spec: auditregv1alpha1.AuditSinkSpec{
+			Policy: auditregv1alpha1.Policy{
+				Level: auditregv1alpha1.LevelRequestResponse,
+				Stages: []auditregv1alpha1.Stage{
+					auditregv1alpha1.StageRequestReceived,
+					auditregv1alpha1.StageResponseStarted,
+					auditregv1alpha1.StageResponseComplete,
+					auditregv1alpha1.StagePanic,
+				},
+			},
+			Webhook: auditregv1alpha1.Webhook{
+				ClientConfig: auditregv1alpha1.WebhookClientConfig{
+					URL: &server1.URL,
+				},
+			},
+		},
+	}
+	sink2 := auditregv1alpha1.AuditSink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test2",
+		},
+		Spec: auditregv1alpha1.AuditSinkSpec{
+			Policy: auditregv1alpha1.Policy{
+				Level: auditregv1alpha1.LevelRequestResponse,
+				Stages: []auditregv1alpha1.Stage{
+					auditregv1alpha1.StageRequestReceived,
+					auditregv1alpha1.StageResponseStarted,
+					auditregv1alpha1.StageResponseComplete,
+					auditregv1alpha1.StagePanic,
+				},
+			},
+			Webhook: auditregv1alpha1.Webhook{
+				ClientConfig: auditregv1alpha1.WebhookClientConfig{
+					URL: &server2.URL,
+				},
+			},
+		},
+	}
+
+	t.Run("one sink", func(t *testing.T) {
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink1)
+		require.NoError(t, err, "failed to create audit sink1")
+		t.Log("created audit sink1")
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events, events %#v not found!", missing)
+	})
+
+	t.Run("two sink", func(t *testing.T) {
+		eventList1.Store(auditinternal.EventList{})
```

Comment:
Please add a comment that you're clearing event list 1 to start fresh for this test case